app/src/test/java/mollie/tictactoe/UIBoardManagerTest.java

```diff
@@ -0,0 +1,71 @@
+package mollie.tictactoe;
+
+import org.junit.Test;
+
+import mollie.tictactoe.testdoubles.ButtonMock;
+import mollie.tictactoe.testdoubles.ContextDummy;
+import mollie.tictactoe.testdoubles.ViewFake;
+import mollie.tictactoe.ui.UIBoardManager;
+import ttt.game.Board;
+import ttt.game.Marks;
+
+import static junit.framework.Assert.assertTrue;
+
+public class UIBoardManagerTest {
+
+    ViewFake mView = new ViewFake(new ContextDummy());
+    private ButtonMock mButton = new ButtonMock(new ContextDummy());
+    private GameHelper mGameHelper = new GameHelper();
+    private Board board = new Board(3);
+
+    @Test
+    public void updatesButtonText() {
+        UIBoardManager.updateUI("X", mButton);
+
+        assertTrue(mButton.setTextWasCalled(1, "X"));
+    }
+
+    @Test
+    public void disablesButton() {
+        UIBoardManager.updateUI("X", mButton);
+
+        assertTrue(mButton.setEnabledWasCalled(1, false));
+    }
+
+    @Test
+    public void doesNotUpdateUIBoardIfGameBoardIsEmpty() {
+        UIBoardManager.updateUIBoard(board, mView);
+
+        assertTrue(mButton.setTextWasCalled(0, "X"));
+    }
+
+    @Test
+    public void updatesBoardToShowAnX() {
+        board = board.placeMark(Marks.X, 4);
+        mView.setButton(mButton);
+
+        UIBoardManager.updateUIBoard(board, mView);
+
+        assertTrue(mButton.setTextWasCalled(1, "X"));
+    }
+
+    @Test
+    public void updatesBoardToShowAnO() {
+        board = board.placeMark(Marks.O, 4);
+        mView.setButton(mButton);
+
+        UIBoardManager.updateUIBoard(board, mView);
+
+        assertTrue(mButton.setTextWasCalled(1, "O"));
+    }
+
+    @Test
+    public void disablesButtonIfMarked() {
+        board = board.placeMark(Marks.O, 4);
```

Comment:
That is the beauty of an Enum vs a String. The compiler will be happy with Mark.O, but will reject Mark.0 (zero) as its not one of the possible enum variants.
On the other hand, both "O" and "0" are valid strings 😄 

As such, I feel like Mark.X and Mark.O are ok in this context